training/train_ellipse.py

```
@@ -0,0 +1,1534 @@
+            raise RuntimeError(f"Model save failed - file not created at {output_path}")
+        print(
+            f"Model saved: {output_path} "
+            f"({os.path.getsize(output_path) / 1024 / 1024:.2f} MB)"
+        )
+
+    # =========================================================================
+    # Data Augmentation
+    # =========================================================================
+    class RandomHorizontalFlip(object):
+        def __call__(self, img, label):
+            if random.random() < 0.5:
```

Ruff (0.14.7)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)